### PR TITLE
179922946-add-delinquent-filtering-in-data-centers

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
add filtering by delinquent and inactive in data centers and asns views

#### How should this be manually tested?
- data center show and asn show pages should have filtering working correctly
- in the index pages (for data centers / asns) delinquent number should be a link to filter

#### What are the relevant tickets?
https://www.pivotaltracker.com/story/show/179922946

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: app/views/data_centers/_data_center_list.html.erb [hunk 1]
```diff
@@ -4,11 +4,14 @@
     <td class="column-sm"><%= dc_data[:aso] %></td>
     <td class="column-md"><%= dc_data[:count] %></td>
     <td class="column-md">
-      <%= dc_data[:delinquent_validators] %>
       <% if dc_data[:delinquent_validators] > 0 %>
-         <small class="text-muted">
+        <%= link_to dc_data[:delinquent_validators], 
+          data_center_url(network: params[:network], key: dc_key.gsub('/', '-slash-'), filter_by: :delinquent) %>
```
Comment (kbiala) [inline app/views/data_centers/_data_center_list.html.erb:9]: Too small indentation

File: test/models/validator_score_v1_test.rb [hunk 2]
```diff
@@ -140,4 +140,45 @@ class ValidatorScoreV1Test < ActiveSupport::TestCase
...
+
+    assert_equal 1, ValidatorScoreV1.filtered_by(:delinquent).count
+    assert ValidatorScoreV1.filtered_by(:delinquent).first.delinquent
+  end
+
+  test 'filter_by when filtered by inactive should return correct results' do
+    create_validators_for_filtering_test
+
+    assert_equal 1, ValidatorScoreV1.filtered_by(:inactive).count
+    refute ValidatorScoreV1.filtered_by(:inactive).first.validator.is_active
```
Comment (radek-polcode) [inline test/models/validator_score_v1_test.rb:155]:
```suggestion
    refute ValidatorScoreV1.filtered_by(:inactive).all? { |score| score.validator.is_active }
```

File: test/models/validator_score_v1_test.rb [hunk 3]
```diff
@@ -140,4 +140,45 @@ class ValidatorScoreV1Test < ActiveSupport::TestCase
       score.update(commission: 20)
     end
   end
+
+  test 'filter_by when filtered by delinquent should return correct results' do
+    create_validators_for_filtering_test
+
+    assert_equal 1, ValidatorScoreV1.filtered_by(:delinquent).count
+    assert ValidatorScoreV1.filtered_by(:delinquent).first.delinquent
```
Comment (radek-polcode) [inline test/models/validator_score_v1_test.rb:148]:
```suggestion
    assert ValidatorScoreV1.filtered_by(:delinquent).all? { |val| val.delinquent }
```

File: app/models/validator_score_v1.rb [hunk 4]
```diff
@@ -88,6 +88,17 @@ def create_commission_history
     CreateCommissionHistoryService.new(self).call
   end
 
+  def self.filtered_by(filter)
+    case filter
+    when :delinquent
+      where(delinquent: true)
+    when :inactive
+      joins(:validator).where('validator.is_active': false)
```
Comment (radek-polcode) [inline app/models/validator_score_v1.rb:96]:
```suggestion
      select('validator_score_v1s.*, validator.is_active').joins(:validator).where('validator.is_active': false)
```

**Review by kbiala — CHANGES_REQUESTED** [review]
Filtering on staging server is not working. It shows all validators in ASN, no matter which filter is active. Seems to be working fine for Data Center views.

File: app/views/asns/show.html.erb [hunk 5]
```diff
@@ -54,9 +54,19 @@
   </div>
 </div>
 
-<%= link_to 'Back to ASN list',
-            data_centers_url(network: params[:network], sort_by: 'asn'),
-            class: 'btn btn-sm btn-secondary mb-4' %>
+<%= link_to 'Filter by delinquent',
+    asn_url(network: params[:network], asn: params[:asn], filter_by: :delinquent),
+    class: "btn btn-sm btn-secondary mb-4 #{'active' if params[:filter_by] == 'delinquent'}" %>
```
Comment (kbiala) [inline app/views/asns/show.html.erb:59]: Please add 'mr-1' class for each button. In the future please try to keep styling consistent with the rest of the page.

**Review by kbiala — CHANGES_REQUESTED** [review]

**Review by kbiala — APPROVED** [review]

**Pull request merged**